tree/ntuple/v7/inc/ROOT/RField.hxx

```diff
@@ -792,6 +794,24 @@ public:
    virtual void AcceptVisitor(Detail::RFieldVisitor &visitor) const;
 }; // class RFieldBase
 
+namespace Internal {
+// At some point, RFieldBase::OnClusterCommit() may allow for a user-defined callback to change the
+// column representation. For now, we inject this for testing and internal use only.
+struct RFieldRepresentationModifier {
+   static void SwapPrimayColumnRepresentation(RFieldBase &field, std::uint16_t newRepresentationIdx)
+   {
+      R__ASSERT(newRepresentationIdx != 0);
+      R__ASSERT(newRepresentationIdx < field.fColumnRepresentatives.size());
+      const auto N = field.fColumnRepresentatives[0].get().size();
+      std::swap(field.fColumnRepresentatives[0], field.fColumnRepresentatives[newRepresentationIdx]);
+      for (std::size_t i = 0; i < N; ++i) {
+         std::swap(field.fColumns[i], field.fColumns[i + newRepresentationIdx * N]);
```

Comment:
This means `fColumns` is not "in order of representation and column index" anymore, as described in the comment above...

Reply:
Fixed in https://github.com/root-project/root/pull/16143/commits/6e34693ef530066c2134dc2b1f979c6525214f72